score.go

```diff
@@ -108,18 +108,32 @@ func (d *indexData) scoreFile(fileMatch *FileMatch, doc uint32, mt matchTree, kn
 	}
 }
 
+func isAtRoot(path string) bool {
+	return strings.IndexRune(path, '/') == -1
+}
+
 // calculateTermFrequency computes the term frequency for the file match.
 //
 // Filename matches count more than content matches. This mimics a common text
 // search strategy where you 'boost' matches on document titles.
-func calculateTermFrequency(cands []*candidateMatch, df termDocumentFrequency) map[string]int {
+func calculateTermFrequency(fm *FileMatch, cands []*candidateMatch, df termDocumentFrequency) map[string]int {
 	// Treat each candidate match as a term and compute the frequencies. For now, ignore case
 	// sensitivity and treat filenames and symbols the same as content.
 	termFreqs := map[string]int{}
+
+	boostFileName := 2
+
+	var evaluated bool
```

Comment:
Why do we need this `evaluated` flag? Could we just do this?

```
		if cand.fileName {
                        boostFileName := 2
			if isAtRoot(fm.FileName) {
			       boostFileName = 5
			}
			termFreqs[term] += boostFileName
		} else {
			termFreqs[term]++
		}
}
```

Reply:
We could, but we only need to calculate `isAtRoot` once because all candidates belong to the same file. There might be candidates that match different parts of the filename, so in your version we might end up calling `isAtRoot` more than we need to.